providers/common/callback_prov.c

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/core.h>
+#include <openssl/core_numbers.h>
+#include "prov/callback.h"
+
+static OSSL_core_generic_callback_fn *c_generic_callback = NULL;
+
+int ossl_prov_callback_from_dispatch(const OSSL_DISPATCH *fns)
+{
+    for (; fns->function_id != 0; fns++) {
```

Comment:
I find it kind of strange that you loop over all the dispatch functions twice. Once here, and once in defltprov.c/fipsprov.c/legacyprov.c

Reply:
Yeah well...  it's essentially to avoid copying the same set of cases all over the place (and potentially getting it wrong somewhere).  It's quite possible that we should just have a central generic *get-me-everything* type of call and be done with it.  I can certainly make such a thing, question is if that should happen in this PR or in a separate one...  it's kind of off-topic for this one...